tests/fake_github.py

```diff
@@ -162,6 +162,14 @@ def close(self, merge=False):
         self.merged = merge
         self.closed_at = datetime.datetime.now()
 
+    def reopen(self):
+        """
+        Re-open a pull request.
+        """
+        self.state = "open"
```

Comment:
@nedbat I got the transitions working by setting this to "reopened", and then being able to use that info within `desired_support_state` to set the right jira status (we need that info in that function to set the new jira state properly). it will fail the `assert anything_happened2 is False` test because the Jira status has changed.

I could either do this pretty simple thing (munge the state if the action is reopened), or, make this test pass with state="open" (I can think of a way to do that, which is within `desired_support_state` for every "open" pr, check to see if there is a SURVEY comment). Either way the `assert anything_happened2 is False` will not be true, because the jira status is changing.

Thoughts on best way?

Reply:
Also I think the state of the PR has always been "open" when the action is "reopened". I can't find an exact reference, but do a command line reopen you just make an api call with state=open.

Reply:
It's fine to change the test "assert anything_happened" things to be correct for the way the code behaves.  The thing to do first is figure out what we want to happen, and then make it happen.  This line in fake_github has to stay as it is, because it emulates GitHub: when a PR is reopened, the state is set to "open."

Reply:
OK, so this comes around to my initial problem of not knowing how to test the actual change I want to make. Sure, this comes in with `state = "open"`. It _also_ comes in with `event["action"] = "reopened"`.

I'm suggesting to intercept that action in `github_views.hook_receiver()` and munge the state to `"reopened"`. 

```
    pr = event["pull_request"]
    pr_number = pr["number"]
    action = event["action"]
    if action == "reopened":
        pr["state"] = "reopened"
```

That way, code in `pr_tracking.desired_support_state` understands this is a different case and can, around line 305, add a clause

```
        if state == "closed":
            desired.jira_status = "Rejected"
        elif state == "merged":
            desired.jira_status = "Merged"
        elif state == "reopened":
            desired.jira_status = "Community Manager Review"
```

But, I don't see a way in the tests to pass the action context. In `fake_github` we mock the PR details, but we don't mock the full event call as far as I can tell. These tests rely on setting _state_ differences, rather than distinguishing between two different actions that caused the PR to get into the same state.

Check out my latest commit. The tests will fail. But if you manually munge the PR state, they pass. So how could I go about leaving the test definition alone, but making sure the new code at github_views L88-89 get executed properly?

Reply:
Hmm I never actually pushed that commit.... one day, we'll get this 😁